study1/week6/LengthOfLongestSubstring.java

```diff
@@ -0,0 +1,36 @@
+import java.lang.Math;
+
+class LengthOfLongestSubstring {
+    public int lengthOfLongestSubstring(String s) {
+
+        Map<Character, Integer> charList = new HashMap<>();
+        int res = 0;
+        int tmpSize = 0;
+
+        for (int cur = 0; cur < s.length(); cur++) {
+
+            if (charList.containsKey(s.charAt(cur))) {
+                res = Math.max(tmpSize, res);
```

Comment:
You can move this after to line 19, as you only need to calculate the max when tmpSize has changed. Then, you don't have to do `Math.max(tmpSize, res)` on line 23, you can do `return res;` instead.